packages/sources/tiingo/src/endpoint/forex.ts

```diff
@@ -37,15 +38,38 @@ interface ResponseSchema {
   midPrice: number
 }
 
+export type TOptions = {
+  url: string
+  inverse?: boolean
+}
+
+const getUrl = (from: string, to: string) => ({
+  url: util.buildUrlPath('/tiingo/fx/:ticker/top', { ticker: `${from}${to}`.toLowerCase() }),
+})
+
+const getIncludesOptions = (
+  //@ts-expect-error no-unused-vars
+  validator: Validator<TInputParameters>,
+  include: IncludePair,
+) => {
+  return {
+    ...getUrl(include.from, include.to),
+    inverse: include.inverse,
+  }
+}
+
 export const execute: ExecuteWithConfig<Config> = async (request, _, config) => {
-  const validator = new Validator(request, inputParameters, {}, { overrides })
+  const validator = new Validator(request, inputParameters, {}, { includes, overrides })
 
   const jobRunID = validator.validated.id
-  const base = validator.overrideSymbol(NAME, validator.validated.data.base)
-  const quote = validator.validated.data.quote
-  const ticker = `${base}${quote}`.toLowerCase()
   const resultPath = (validator.validated.data.resultPath || '').toString()
-  const url = util.buildUrlPath('/tiingo/fx/:ticker/top', { ticker })
+
+  const { url, inverse } = util.getPairOptions<TOptions, TInputParameters>(
+    AdapterName,
+    validator,
+    getIncludesOptions,
+    getUrl,
+  ) as TOptions // If base and quote cannot be batched, getPairOptions will return TOptions
```

Comment:
I wonder if we can improve this. I don't like needing to coerce the type here.
It was introduced [here](https://github.com/smartcontractkit/external-adapters-js/pull/2028/files#diff-db405e686a22de740fe2caac9fa7e96ac17b56ab1ff5fa2ff40d49453a025fa2R601). I've reached out to Austin for context.

Reply:
This may be better as 2 distinct methods, like `getPairOptions` and a wrapper `getBatchedPairOptions` (or distinct methods with common helpers). That would prevent the coercion here.

Reply:
We could probably also do this with a [conditional type](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html) based on the input. Amit would you like to flex some Typescript muscles and take a stab at this?

[Here](https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/core/bootstrap/src/lib/modules/requester.ts#L388) is an example where I have done this before in the codebase.

Reply:
Sure I'll take a shot at it!

Reply:
After some discussions, we decided to go with the 2 disctinct methods approach. I updated Amberdata, Kaiko, Openexchangerates, and Tiingo accordingly